src/amo/components/HomepageShelves/index.js

```diff
@@ -0,0 +1,82 @@
+/* @flow */
+import * as React from 'react';
+import config from 'config';
+
+import translate from 'amo/i18n/translate';
+import LandingAddonsCard from 'amo/components/LandingAddonsCard';
+import {
+  INSTALL_SOURCE_FEATURED,
+  INSTALL_SOURCE_FEATURED_COLLECTION,
+  LANDING_PAGE_EXTENSION_COUNT,
+  LANDING_PAGE_THEME_COUNT,
+} from 'amo/constants';
+import type { ResultShelfType } from 'amo/reducers/home';
+
+import './styles.scss';
+
+export const MOZILLA_USER_ID = config.get('mozillaUserId');
+
+type Props = {|
+  i18n: Object,
+  loading: boolean,
+  placeholderCount: number,
+  shelves: Array<ResultShelfType>,
+|};
+
+class HomepageShelves extends React.Component<Props> {
+  static defaultProps = {
+    placeholderCount: LANDING_PAGE_EXTENSION_COUNT,
+  };
+
+  render() {
+    const { i18n, loading, placeholderCount, shelves } = this.props;
+
+    const homepageShelves: Array<React.Node> = shelves.map((shelf) => {
+      const { addons, criteria, endpoint, footer, title } = shelf;
+      const titleStr = title.toString();
+      const header = titleStr.replace(/\s/g, '');
```

Comment:
It might make it more readible if you were to add a dash (`-`) instead of an empty string.